app/svelte/src/client/preview/render.js

```diff
@@ -0,0 +1,52 @@
+import { document } from 'global';
+import { stripIndents } from 'common-tags';
+
+function mountView({ Component, target, data, on }) {
+  const component = new Component({ target, data });
+
+  if (on) {
+    // Attach svelte event listeners.
+    Object.keys(on).forEach(eventName => {
+      component.on(eventName, on[eventName]);
```

Comment:
Are those listeners automatically detached when component unmounts?

Reply:
I will look into it. I know some are, but they are internally managed within the components.

Reply:
Svelte removes those listeners... but the component has to be `destroy()`-ed.
Usually, svelte will take care of calling `destroy()` on components when they are removed from the DOM.
However, I'm not sure if you need to manually `destroy()` them when switching stories in this context

Reply:
Yeah, svelte will not remove any listeners added manually using `component.on()`.
They need to be removed manually.

```js
//add listener
const listener = component.on('someEvent', function(){});

//remove listener
listener.cancel();
```

Reply:
For now I am not calling destroy on the component when changing stories, as the JS reference to it should be garbage collected, and the DOM nodes are removed by storybook.

I am adding something similar to what the Vue client/preview does, which is to keep a reference to the listeners and then cancel them when running a new render().

Thanks @ekhaled for the info.

Reply:
We decided on destroying the components rather than cancelling the listeners.